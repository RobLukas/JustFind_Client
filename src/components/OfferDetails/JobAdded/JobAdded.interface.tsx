interface JobAddedProps {
  createdAt: Date;
}

export default JobAddedProps;
