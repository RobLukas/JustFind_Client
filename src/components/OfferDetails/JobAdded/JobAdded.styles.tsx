import styled from 'styled-components';

const NewBar = styled.span(({ theme }) => ({
  '@media (max-width: 1020px)': {
    marginLeft: 'auto',
  },
  color: theme.styles.offers.newTitle.color,
  backgroundColor: theme.styles.offers.newTitle.background,
  padding: '4px 7px',
  borderRadius: 16,
  marginLeft: 20,
  whiteSpace: 'nowrap',
  fontWeight: 400,
  verticalAlign: 'bottom',
  fontSize: 12,
}));

export default {
  NewBar,
};
