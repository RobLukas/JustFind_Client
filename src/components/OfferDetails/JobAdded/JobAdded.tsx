import React, { FC } from 'react';
import S from './JobAdded.styles';
import TileRoundedEdges from '../TileRoundedEdges/TileRoundedEdges';
import JobAddedProps from './JobAdded.interface';

const JobAdded: FC<JobAddedProps> = ({ createdAt }: JobAddedProps) => {
  const toFiveDaysIsNew = 5;
  const timeDifferenceFromNow = new Date().getDate() - createdAt.getDate();
  const isNew = timeDifferenceFromNow < toFiveDaysIsNew;

  return (
    <>
      {isNew ? (
        <S.NewBar>New</S.NewBar>
      ) : (
        <TileRoundedEdges>{`${timeDifferenceFromNow} ago`}</TileRoundedEdges>
      )}
    </>
  );
};

export default JobAdded;
