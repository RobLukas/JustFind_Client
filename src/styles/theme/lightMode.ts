export default {
  name: 'Light Theme',
  styles: {
    primary: {
      background: '#f3f6f8',
      color: '#37474f',
    },
    secondary: {
      background: '#fff',
      color: '#99a1ab',
    },
    dropdown: {
      button: {
        hover: '#f3f6f8',
      },
    },
    scrollbar: {
      thumbBackground: '#c1c1c1',
      background: 'rgba(0,0,0,0.07)',
      hover: 'rgba(0,0,0,0.09)',
    },
    filter: {
      button: {
        color: '#777',
      },
      selected: {
        background: 'linear-gradient(to right, #ba68c8, #9c27b0)',
        color: '#fff',
      },
    },
    offer: {
      header: {
        title: '#fff',
        filter: {
          brightness: '1',
        },
        icons: {
          companyName: '#ff5252',
          companySize: '#fb8c00',
          companyEMPType: '#ab47bc',
          companySkill: '#66bb6a',
          new: '#448aff',
        },
      },
      desc: {
        seperateLine: {
          color: '#f3f6f8',
        },
        skillPoint: {
          selected: '#ff4081',
          unselected: '#393939',
        },
      },
    },
    offers: {
      newTitle: {
        background: '#d9ddfc',
        color: '#4256ef',
      },
      techTitle: {
        background: '#fff',
        border: '#e0e0e0',
      },
    },
  },
};
