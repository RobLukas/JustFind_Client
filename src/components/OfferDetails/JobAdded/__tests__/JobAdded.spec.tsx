import React from 'react';
import { cleanup, render } from '@testing-library/react';

import JobAddedProps from '../JobAdded.interface';
import JobAdded from '../JobAdded';
import { ThemeProvider } from 'styled-components';
import lightMode from '@styles/theme/lightMode';

describe('<JobAdded />', () => {
  afterEach(cleanup);
  it('render correctly with New annotation', () => {
    const props: JobAddedProps = { createdAt: new Date() };
    const { container, getByText } = render(
      <ThemeProvider theme={lightMode}>
        <JobAdded {...props}></JobAdded>
      </ThemeProvider>,
    );
    expect(getByText('New')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
  it('render correctly with notify 5d ago', () => {
    const date = new Date();
    const dateFiveDaysAgo = date.setDate(date.getDate() - 5);

    const props: JobAddedProps = { createdAt: new Date(dateFiveDaysAgo) };
    const { container, getByText } = render(
      <ThemeProvider theme={lightMode}>
        <JobAdded {...props}></JobAdded>
      </ThemeProvider>,
    );
    expect(getByText('5d ago')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
