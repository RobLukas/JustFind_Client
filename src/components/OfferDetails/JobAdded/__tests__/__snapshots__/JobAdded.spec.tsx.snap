// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobAdded /> render correctly with New annotation 1`] = `
<span
  class="sc-bdnylx AGibd"
>
  New
</span>
`;

exports[`<JobAdded /> render correctly with notify 5d ago 1`] = `
<span
  class="sc-gtssRu glfUpC"
>
  5d ago
</span>
`;
